src/plugins/controls/public/control_group/embeddable/control_group_chaining_system.ts

```diff
@@ -35,6 +40,38 @@ interface ChainingSystem {
   onChildChange: (props: OnChildChangedProps) => void;
 }
 
+export interface ChildEmbeddableOrderCache {
+  IdsToOrder: { [key: string]: number };
+  idsInOrder: string[];
+  lastChildId: string;
+}
+
+const getOrdersFromPanels = (panels?: ControlsPanels) => {
+  return Object.values(panels ?? {}).map((panel) => ({
+    id: panel.explicitInput.id,
+    order: panel.order,
+  }));
+};
+
+export const controlOrdersAreEqual = (panelsA?: ControlsPanels, panelsB?: ControlsPanels) =>
+  deepEqual(getOrdersFromPanels(panelsA), getOrdersFromPanels(panelsB));
+
+export const cachedChildEmbeddableOrder = memoize(
+  (panels: ControlsPanels) => {
+    const IdsToOrder: { [key: string]: number } = {};
+    const idsInOrder: string[] = [];
+    Object.values(panels)
+      .sort((a, b) => (a.order > b.order ? 1 : -1))
+      .forEach((panel) => {
+        IdsToOrder[panel.explicitInput.id] = panel.order;
+        idsInOrder.push(panel.explicitInput.id);
```

Comment:
Hahah yeah this is just a caching layer to avoid having to recalculate the order, so I also introduced `IdsToOrder` to avoid looping through them... I tend to err on the side of "use a little more memory to avoid using CPU cycles", but sometimes that kinda gets in the way of readability.

I would definitely add a comment or change the name... Maybe just a comment that says `//cache used to avoid looping through panels to find the order` or something similar. What do you think?